token/services/identity/role/role.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package role
+
+import (
+	"runtime/debug"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/hash"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/identity"
+	idriver "github.com/hyperledger-labs/fabric-token-sdk/token/services/identity/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/logging"
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+)
+
+type localMembership interface {
+	DefaultNetworkIdentity() driver.Identity
+	IsMe(id driver.Identity) bool
+	GetIdentityInfo(label string, auditInfo []byte) (idriver.IdentityInfo, error)
+	GetIdentifier(id driver.Identity) (string, error)
+	GetDefaultIdentifier() string
+	RegisterIdentity(config driver.IdentityConfiguration) error
+	IDs() ([]string, error)
+}
+
+// Role models a role whose identities are anonymous
+type Role struct {
+	logger          logging.Logger
+	roleID          identity.RoleType
+	networkID       string
+	localMembership localMembership
+	nodeIdentity    driver.Identity
+}
+
+func NewRole(logger logging.Logger, roleID identity.RoleType, networkID string, nodeIdentity driver.Identity, localMembership localMembership) *Role {
+	return &Role{
+		logger:          logger,
+		roleID:          roleID,
+		networkID:       networkID,
+		localMembership: localMembership,
+		nodeIdentity:    nodeIdentity,
+	}
+}
+
+func (r *Role) ID() identity.RoleType {
+	return r.roleID
+}
+
+// GetIdentityInfo returns the identity information for the given identity identifier
+func (r *Role) GetIdentityInfo(id string) (idriver.IdentityInfo, error) {
+	if r.logger.IsEnabledFor(zapcore.DebugLevel) {
```

Comment:
`IsEnabledFor` is not needed unless you're doing some complex computation

Reply:
okay